### PR TITLE
Implementing Random Forest with clustering only true activations

### PR DESCRIPTION
**Pull request by stubbornfox**
run train.py to train
run prediction_path.py to see the graph

### CODE REVIEW

File: steps/s2_interpret_segments.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def interpret_segments(configuration: Configuration, dataset: Dataset) -> None:
             activation_file = activation_path(f'{image_id}.npz')
             prediction_file = prediction_path(f'{image_id}.npz')
 
-            if exists(activation_file) and exists(prediction_file):
+            if exists(activation_file) or exists(prediction_file):
```
Comment (NiclasvanEyk) [inline steps/s2_interpret_segments.py]: why was this change made? If the activation file exists, but the prediction file does not, then no accuracy can be calculated for the clusters / segments and we definitely have to recompute them, right?
So we can only skip an image if _both_ file exists, only one of them is not enough.

File: utils/configuration.py [hunk 2]
```diff
@@ -31,3 +33,6 @@ class Configuration:
 
     # Either 'ntsnet' (default), 'resnet256' or 'resnet128'
     network = 'ntsnet'
+
+    # This code only support meaning activation cluster
```
Comment (NiclasvanEyk) [inline utils/configuration.py]: Why do we need an entry in the config, if this is the only valid option?

File: utils/configuration.py [hunk 3]
```diff
@@ -17,6 +17,8 @@ class Configuration:
     # contain in order to be seen as meaningful.
     cluster_accuracy_threshold: int
 
+    max_depth: int
```
Comment (NiclasvanEyk) [inline utils/configuration.py:21]: Could you document this a bit? I guess it is related to the tree?

**Review by NiclasvanEyk — CHANGES_REQUESTED** [review]
Nice usage of the emojis for the edges 😅

File: steps/s2_interpret_segments.py (hunk 1 above)
Comment (stubbornfox) [reply steps/s2_interpret_segments.py]: My bad, for some reason I don't have prediction files on the test set images(I don't want to run the CNN again for the test)

File: utils/configuration.py (hunk 2 above)
Comment (stubbornfox) [reply utils/configuration.py]: I have a plan to support both kinds of choosing the important cluster method, but actually not the case. If you are ok with only cluster using the meaning activation then I will remove that config

File: utils/configuration.py (hunk 3 above)
Comment (stubbornfox) [reply utils/configuration.py:21]: Add description

**Pull request merged**